url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/step_out.java

```diff
@@ -0,0 +1,220 @@
+                soutLine = new String();
+                ScanFile.pos = ScanFile.br.getFilePointer();
+                initial_pos = ScanFile.pos.intValue();
+                Register.r[16].b = ScanFile.pos.intValue();
+//                br.seek(initial_pos);
+                ScanFile.current_line = ScanFile.getLineNumber(FrontEnd.activepane, initial_pos);
+            }
+            while (ScanFile.call_count >= exitCondition && (soutLine = ScanFile.br.readLine()) != null)
+            {
+                ScanFile.pos = (Long) ScanFile.br.getFilePointer();
+
+                (new handlers()).update(handlers.regMode);
+                Register.r[16].b = ScanFile.pos.intValue();
+                initial_pos = ScanFile.pos.intValue();
+                ScanFile.current_line = ScanFile.getLineNumber(FrontEnd.activepane, initial_pos);
+                soutLine = soutLine.toUpperCase();
+                RemoveComments.setflag(StepRun.commentflag);
+                soutLine = RemoveComments.RC(soutLine);
+                StepRun.commentflag = RemoveComments.getcommetnflag();
+                if (soutLine.contains(":"))
+                {
+                    StringTokenizer st3 = new StringTokenizer(soutLine, ":");
+                    int ct_label = st3.countTokens();
+                    if (ct_label == 1)
+                    {
+                        continue;
+                    } else
+                    {
+                        if (ct_label == 2)
+                        {
+                            soutLine = st3.nextToken();
+                            soutLine = st3.nextToken();
+                        } else
+                        {
+                            while (st3.hasMoreTokens())
+                            {
+                                soutLine += st3.nextToken();
+                            }
+                        }
+                        soutLine = soutLine.trim();
+                        if (!soutLine.isEmpty())
+                        {
+                            ScanFile.tokenizeInstruction(soutLine);
+                            if (frontend.FrontEnd.exceptionraised == 0)
+                        {
+                             FrontEnd.appendToPane(FrontEnd.statuswindow,"Instruction ",Color.GREEN);
+                                    FrontEnd.appendToPane(FrontEnd.statuswindow,ScanFile.stepins,Color.BLACK);
+                                    FrontEnd.appendToPane(FrontEnd.statuswindow, " Executed Successfully \n",Color.GREEN);
+                        } else
+                        {
+                             FrontEnd.appendToPane(FrontEnd.statuswindow,"Instruction ",Color.RED);
+                                    FrontEnd.appendToPane(FrontEnd.statuswindow,ScanFile.stepins,Color.BLACK);
+                                    FrontEnd.appendToPane(FrontEnd.statuswindow, " Failed To Execute \n",Color.RED);
+                        }
+                        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 147:161, 164:180.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.